hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiryManager.java

```diff
@@ -37,78 +37,161 @@
 import java.util.function.Supplier;
 
+import static java.util.Comparator.comparing;
+
+/**
+ * Manager of two queues of expiration events---one for payer records, one for schedule entities.
+ *
+ * There are two management responsibilities:
+ * <ol>
+ *    <li>On restart or reconnect, rebuild the expiration queues from state.</li>
+ *    <li>At the first consensus second an entity is expired, remove it from its parent collection.</li>
+ * </ol>
+ */
 public class ExpiryManager {
+	private final long shard, realm;
+
 	private final RecordCache recordCache;
+	private final ScheduleStore scheduleStore;
 	private final Map<TransactionID, TxnIdRecentHistory> txnHistories;
+	private final Supplier<FCMap<MerkleEntityId, MerkleAccount>> accounts;
 	private final Supplier<FCMap<MerkleEntityId, MerkleSchedule>> schedules;
 
-	private final ScheduleStore scheduleStore;
-
-	long sharedNow;
-	MonotonicFullQueueExpiries<Long> payerExpiries = new MonotonicFullQueueExpiries<>();
-	MonotonicFullQueueExpiries<Pair<Long, Consumer<EntityId>>> entityExpiries = new MonotonicFullQueueExpiries<>();
+	private final MonotonicFullQueueExpiries<Long> payerRecordExpiries =
+			new MonotonicFullQueueExpiries<>();
+	private final MonotonicFullQueueExpiries<Pair<Long, Consumer<EntityId>>> shortLivedEntityExpiries =
+			new MonotonicFullQueueExpiries<>();
 
 	public ExpiryManager(
 			RecordCache recordCache,
-			Map<TransactionID, TxnIdRecentHistory> txnHistories,
 			ScheduleStore scheduleStore,
+			HederaNumbers hederaNums,
+			Map<TransactionID, TxnIdRecentHistory> txnHistories,
+			Supplier<FCMap<MerkleEntityId, MerkleAccount>> accounts,
 			Supplier<FCMap<MerkleEntityId, MerkleSchedule>> schedules
 	) {
+		this.accounts = accounts;
+		this.schedules = schedules;
 		this.recordCache = recordCache;
 		this.txnHistories = txnHistories;
 		this.scheduleStore = scheduleStore;
 
-		this.schedules = schedules;
+		this.shard = hederaNums.shard();
+		this.realm = hederaNums.realm();
+	}
+
+	/**
+	 * Purges any references to expired entities (at this time, records or schedules).
+	 *
+	 * @param now the consensus second
+	 */
+	public void purge(long now) {
+		purgeExpiredRecordsAt(now);
+		purgeExpiredShortLivedEntities(now);
 	}
 
-	public void trackRecord(AccountID owner, long expiry) {
-		payerExpiries.track(owner.getAccountNum(), expiry);
+	/**
+	 * Begins tracking an expiration event.
+	 *
+	 * @param event the expiration event to track
+	 * @param expiry the earliest consensus second at which it should fire
+	 */
+	public void trackExpirationEvent(Pair<Long, Consumer<EntityId>> event, long expiry) {
+		shortLivedEntityExpiries.track(event, expiry);
 	}
 
-	public void restartTrackingFrom(FCMap<MerkleEntityId, MerkleAccount> accounts) {
+	/**
+	 * When payer records are in state (true by default), upon restart or reconnect the
+	 * expiry manager needs to rebuild its expiration queue so it can correctly purge
+	 * these records as their 180s lifetimes expire.
```

Comment:
nit: add `(default ttl)` to explain the 180s magic number

Reply:
Done.